lib/ansible/cli/scripts/ansible_cli_stub.py

```diff
@@ -100,6 +100,11 @@ def error(self, msg, wrap_text=None):
             else:
                 raise
 
+        b_ansible_dir = os.path.expanduser(os.path.expandvars(b"~/.ansible"))
```

Comment:
Should this use `os.path.dirname(C.GALAXY_TOKEN_PATH)` so it acts correctly if this configuration variable is changed and the default `~/.ansible/` is not actually the correct location to create? And, if so, should it use `os.makedirs()` in case there are multiple levels configured in the token path that might need to be created?

Reply:
I originally had this that way but after talking with @abadger he informed me of a security implications of doing that. The trouble with that is if a user has changed the path to something like `/var/tmp/totally_secure/.ansible`. If a malicious user knows about this dir they could redirect where the files get written using a symlink and will now be able to read your token. We can assume on Linux that `~/[anything]` is secure so are happy to hardcode that path so our defaults work properly. But as soon as a user changes that path it is up to them to ensure that 1. the directory exists, 2. the directory is secure and owned by them.

So ultimately this change will ensure that the default of `C.GALAXY_TOKEN_PATH` works without the user having the create the directory but if a user changes the path then they need to ensure the dir is present and is secure.